WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -201,6 +201,19 @@ extension CoreDataStack {
         }
     }
 
+    /// Perform a query using the `mainContext` and return the result.
+    ///
+    /// This is a re-implemenation of https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/3802019-performandwait,
+    /// which is available from iOS 15.0 (hence the deprecation attribute below).
+    @available(iOS, deprecated: 15.0, message: "Use `mainContext.performAndWait` instead.")
+    func performQuery<T>(_ block: @escaping (NSManagedObjectContext) -> T) -> T {
+        var value: T! = nil
```

Comment:
At first sight, I didn't make sense to me. I though, "_How can assigning `nil` to a IUO work?!_" Then, I remembered reading about some under the hood changes to IUO some time ago and found the [this article on `swift.org`](https://www.swift.org/blog/iuo/). 

> Implicitly unwrapped optionals have been reimplemented such that they are no longer a distinct type from `Optional<T>`. As a result, type checking is more consistent and there are fewer special cases in the compiler. Removing these special cases should lead to fewer bugs in handling of these declarations.

No that makes more sense 😄 

Trying the code in the REPL confirms it:

<img width="1165" alt="image" src="https://user-images.githubusercontent.com/1218433/213033113-101c9487-de5e-4898-85aa-b0796f35c62d.png">

Reply:
Given the fact that this method is defined on the stack, and not on `ManagedObjectContext`, I wonder if we should deprecate it after all? 

It can still be useful for isolating the underlying `mainContext` if that's what we're after? Otherwise, should we define it as an extension on `ManagedObjectContext` to make the refactor more straightforward later?

Reply:
> It can still be useful for isolating the underlying mainContext if that's what we're after?

Agreed. I think it's nice to have `performQuery` for reading and `performAndSave` for writing. I've removed the deprecation attribute in https://github.com/wordpress-mobile/WordPress-iOS/pull/19910/commits/139cc5789b153b9110a0f3a3f1c8b07b75c60a97